### PR TITLE
Feature/bib 521 print dialog and enter 09 2020

### PR DESCRIPTION
**Pull request by sinejespersen**

### CODE REVIEW

File: assets/js/steps/check-in-items.js [hunk 1]
```diff
@@ -22,6 +22,8 @@ import Input from './components/input';
 import { adaptListOfBooksToBanner } from './utils/banner-adapter';
 import { faBook } from '@fortawesome/free-solid-svg-icons';
 import NumPad from './utils/num-pad';
+import bookStatus from './utils/book-status';
```
Comment (tuj) [inline assets/js/steps/check-in-items.js:25]: BookStatus

File: assets/js/steps/utils/print.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * @constructor
+ */
+function Print({ book }) {
+    /**
+     * Print on load
+     */
+    useEffect(() => {
+        window.print();
+    }, []);
+    return (
```
Comment (tuj) [inline assets/js/steps/utils/print.js:24]: new line before return

File: assets/js/steps/utils/banner-adapter.js [hunk 3]
```diff
@@ -33,6 +33,11 @@ export function adaptListOfBooksToBanner(listOfBooks) {
                 if (book.author) {
                     displayInfo.text = `af ${book.author}`;
                 }
+                if (book.message === 'Reserveret') {
+                    displayInfo.status = bookStatus.RESERVED;
```
Comment (tuj) [inline assets/js/steps/utils/banner-adapter.js:37]: BookStatus

File: assets/js/steps/components/navbar.js [hunk 4]
```diff
@@ -31,28 +32,33 @@ function NavBar({ actionHandler }) {
...
+            color: 'purple',
             data: { flow: 'checkInItems' },
             label: 'Aflever',
             icon: faBook
         }
     ];
 
+    /**
+     * Prints the page, available in status component.
+     *
```
Comment (tuj) [inline assets/js/steps/components/navbar.js:56]: Remove empty line

File: assets/js/steps/components/banner.js [hunk 5]
```diff
@@ -23,12 +23,13 @@ import {
...
  * @constructor
  */
-function Banner({ item }) {
-    let classes = 'banner ';
+function Banner({ item, visibleOnPrint = false }) {
+    let classes = visibleOnPrint ? 'banner visibe-on-print ' : 'banner ';
     let { text, title, status, itemIdentifier } = item;
     let icon = null;
     switch (status) {
         case bookStatus.ERROR:
```
Comment (tuj) [inline assets/js/steps/components/banner.js:31]: BookStatus

File: assets/js/steps/check-in-items.js [hunk 6]
```diff
@@ -77,6 +77,7 @@ function CheckInItems({ actionHandler }) {
     if (context.machineState.get.items) {
         items = adaptListOfBooksToBanner(context.machineState.get.items);
     }
+    const reservedBookForPrint = items.filter(book => book.status === bookStatus.RESERVED).pop();
```
Comment (tuj) [inline assets/js/steps/check-in-items.js:80]: New line before this const

Why only one book?

**Review by tuj — CHANGES_REQUESTED** [review]

File: assets/js/steps/check-in-items.js (hunk 6 above)
Comment (sinejespersen) [reply assets/js/steps/check-in-items.js:80]: Fordi der skal kun printes dén bog som _lige_ er blevet afleveret som er reserveret.

Comment (tuj) [reply assets/js/steps/check-in-items.js]: Do you know this is always the last element of the filtered array?

**Pull request closed without merging**